packages/commonwealth/client/scripts/views/pages/CommunityGroupsAndMembers/Groups/common/GroupForm/AllowList.tsx

```diff
@@ -0,0 +1,272 @@
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [searchFilters.groupFilter, members?.pages, currentPage, groups]);
+
+  useEffect(() => {
+    // Invalidate group memberships cache
+    void queryClient.cancelQueries([ApiEndpoints.FETCH_GROUPS]);
+    void queryClient.refetchQueries([ApiEndpoints.FETCH_GROUPS]);
+  }, []);
+
+  const handleCheckboxChange = (address: string) => {
+    if (allowedAddresses.includes(address)) {
+      setAllowedAddresses((prevItems) =>
+        prevItems.filter((item) => item !== address),
+      );
+    } else {
+      setAllowedAddresses((prevItems) => [...prevItems, address]);
+    }
+  };
+
+  const extraColumns = (member: Member) => {
+    return {
+      address: {
+        sortValue: member.address,
+        customElement: (
+          <div className="table-cell">
+            {formatAddressCompact(member.address)}
+          </div>
+        ),
+      },
+    };
+  };
+
+  return (
+    <section className="form-section">
+      <div className="header-row">
+        <CWText type="h4" fontWeight="semiBold">
+          Allow List
+        </CWText>
+        <CWText type="b2">
+          You can bypass the conditions and add members directly to the group
+          using the table below
+        </CWText>
+      </div>
+
+      <div className="header-column">
+        <CWText type="h5" fontWeight="semiBold">
+          Filter & Search
+        </CWText>
+        <Select
+          options={[
+            ...baseFilterOptions,
+            ...(groups || []).map((group) => {
+              return {
+                label: group.name,
+                value: group.id,
+              };
+            }),
+          ]}
+          placeholder={baseFilterOptions[1].label}
+          onSelect={async (option) => {
+            await handleChange(null, 1);
+            setSearchFilters((g) => ({
+              ...g,
+              groupFilter: (
+                option as {
+                  value: number | BaseGroupFilter;
+                }
+              ).value as string,
+            }));
+          }}
+          selected={searchFilters.groupFilter.toString()}
+        />
+        <CWTextInput
+          fullWidth={true}
+          placeholder="Search members"
+          iconLeft={<MagnifyingGlass size={24} weight="regular" />}
+          onInput={(e) =>
+            setSearchFilters((g) => ({
+              ...g,
+              searchText: e.target.value?.trim(),
+            }))
+          }
+        />
+      </div>
+      <MembersSection
+        filteredMembers={formattedMembers}
+        isLoadingMoreMembers={isLoadingMembers}
+        tableState={tableState}
+        extraColumns={extraColumns}
+        selectedAccounts={allowedAddresses}
+        handleCheckboxChange={handleCheckboxChange}
+      />
+      <div className="pagination-buttons">
+        <CWPagination totalCount={totalPages} onChange={handleChange} />
+      </div>
+    </section>
+  );
+};
+
+export default AllowList;
```

Comment:
create directory Allowlist and inside Allowslist.tsx + index.ts + Allowlist.scss if you use some styling